text/0517-io-os-reform.md

```diff
@@ -483,7 +483,72 @@ throughout IO, we can go on to explore the modules in detail.
 ### `std::env`
 [std::env]: #stdenv
 
-> To be added in a follow-up PR.
+Most of what's available in `std::os` today will move to `std::env`,
+and the signatures will be updated to follow this RFC's
+[Design principles] as follows.
+
+**Arguments**:
+
+* `args`: change to yield an iterator rather than vector if possible; in any case, it should produce an `OsStrBuf`.
```

Comment:
@alexcrichton The effect on the API is something like `Iterator<Item = CowOsString>` rather than `OsString`, I think. (I believe that we can safely avoid copying on Linux at least.)

That said, it's not clear to me that using Cow here is really worthwhile; as @alexcrichton said above, this seems extremely unlikely to be used in an inner loop and it's trivially cacheable if so.